### ISSUE TITLE
Is this plugin still needed with JSPM 0.16 beta

### ISSUE THREAD

**Issue by ifyio** [frankwallis/plugin-typescript#35]
The JSPM 0.16 beta added support for transpiling with typescript. I somehow assumed that it would no longer be necessary to have to install plugin-typescript manually but that doesn't seem to be the case. It seems like I still need to install this plugin. Is this correct or am I missing something?

**Comment by frankwallis** [comment]
You're correct, you don't need to install this plugin to run typescript code in 0.16 beta. However the built-in transpiler does not support type-checking. Also see #18 - the long-term plan for jspm is to get rid of the built-in transpilers and have everything done via plugins.

**Comment by ifyio** [comment]
Ok I see. I didn't realise I had to add the add a 'package' configuration to `config.js` to make it work with SystemJS. I have been playing around with `plugin-typescript` and I have a question. Are self executing bundles the only bundling mechanism that is currently supported? When I do `jspm bundle app bundles/bundle.js --inject`, the app no longer works and no error is reported.

**Comment by frankwallis** [comment]
I see the same problem in the example app, the entry file gets passed to plugin-typescript as an empty string which is why nothing happens. It's not clear to me where the problem lies at the moment but I don't think that the plugin should be called at all as the code has already been bundled.

**Comment by frankwallis** [comment]
More info, the empty string comes from SystemJS loadFromBundle:

```
/*
  System bundles

  Allows a bundle module to be specified which will be dynamically
  loaded before trying to load a given module.

  For example:
  System.bundles['mybundle'] = ['jquery', 'bootstrap/js/bootstrap']

  Will result in a load to "mybundle" whenever a load to "jquery"
  or "bootstrap/js/bootstrap" is made.

  In this way, the bundle becomes the request that provides the module
*/

(function() {
  // bundles support (just like RequireJS)
  // bundle name is module name of bundle itself
  // bundle is array of modules defined by the bundle
  // when a module in the bundle is requested, the bundle is loaded instead
  // of the form System.bundles['mybundle'] = ['jquery', 'bootstrap/js/bootstrap']
  hookConstructor(function(constructor) {
    return function() {
      constructor.call(this);
      this.bundles = {};
      this.loadedBundles_ = {};
    };
  });

  function loadFromBundle(loader, bundle) {
    return Promise.resolve(loader.normalize(bundle))
    .then(function(normalized) {
      loader.loadedBundles_[normalized] = true;
      loader.bundles[normalized] = loader.bundles[normalized] || loader.bundles[bundle];
      return loader.load(normalized);
    })
    .then(function() {
      return '';
    });
  }
```

**Comment by frankwallis** [comment]
And disabling plugin-typescript by changing the loader configuration has no effect.

**Comment by frankwallis** [comment]
@guybedford - We are seeing an issue with ```jspm bundle --inject```. The issue is that the initial load triggers the bundle to get loaded, but then the initial load continues through the pipeline with the source set to an empty string. If the file is part of a bundle shouldn't the load get aborted somehow and just allow the bundle to register the module?

**Comment by guybedford** [comment]
@frankwallis yes an empty string source will pass through the pipeline, but then it will be picked up from the `System.defined` bundle registry on instantiate return as if it was loaded properly to begin with.

If it's not being picked up that is most likely due to the bundle names not matching right. Make sure that the plugin is outputting anonymous `System.register` statements, for the loader and builder to name itself. Also there was a file extension bundling issue in https://github.com/systemjs/systemjs/issues/602, which may be affecting this.

**Comment by frankwallis** [comment]
@guybedford, @ifyio - thanks, yes this does seem to be working now (SystemJS 0.18.6). However it is still loading the plugin when it is not needed as all the typescript has already been compiled inside the bundle?

**Comment by guybedford** [comment]
@frankwallis thanks I've posted https://github.com/systemjs/systemjs/issues/670 to track this issue. In the mean time, including the bundle manually right after SystemJS should work here fine though.

**Comment by frankwallis** [comment]
This is now working in jspm@0.16.0
